extension/src/background/messageListener/freighterApiMessageListener.ts

```diff
@@ -105,94 +105,109 @@ export const freighterApiMessageListener = (
     const { networkUrl } = await getNetworkDetails();
     const isExperimentalModeEnabled = await getIsExperimentalModeEnabled();
     const SDK = isExperimentalModeEnabled ? SorobanSdk : StellarSdk;
-    const transaction = SDK.TransactionBuilder.fromXDR(
-      transactionXdr,
-      networkPassphrase || SDK.Networks[network],
-    );
 
     const { tab, url: tabUrl = "" } = sender;
     const domain = getUrlHostname(tabUrl);
     const punycodedDomain = getPunycodedDomain(domain);
 
     const allowListStr = (await localStore.getItem(ALLOWLIST_ID)) || "";
     const allowList = allowListStr.split(",");
-
     const isDomainListedAllowed = await isSenderAllowed({ sender });
 
-    const directoryLookupJson = await cachedFetch(
-      STELLAR_EXPERT_BLOCKED_ACCOUNTS_URL,
-      CACHED_BLOCKED_ACCOUNTS_ID,
-    );
-    const accountData = directoryLookupJson?._embedded?.records || [];
-
-    const _operations =
-      transaction._operations || transaction._innerTransaction._operations;
-
-    const flaggedKeys: FlaggedKeys = {};
-
-    const isValidatingMemo = (await getIsMemoValidationEnabled()) && isMainnet;
-    const isValidatingSafety =
-      (await getIsSafetyValidationEnabled()) && isMainnet;
-
-    if (isValidatingMemo || isValidatingSafety) {
-      _operations.forEach((operation: { destination: string }) => {
-        accountData.forEach(
-          ({ address, tags }: { address: string; tags: Array<string> }) => {
-            if (address === operation.destination) {
-              let collectedTags = [...tags];
-
-              /* if the user has opted out of validation, remove applicable tags */
-              if (!isValidatingMemo) {
-                collectedTags.filter(
-                  (tag) => tag !== TRANSACTION_WARNING.memoRequired,
-                );
-              }
-              if (!isValidatingSafety) {
-                collectedTags = collectedTags.filter(
-                  (tag) => tag !== TRANSACTION_WARNING.unsafe,
-                );
-                collectedTags = collectedTags.filter(
-                  (tag) => tag !== TRANSACTION_WARNING.malicious,
-                );
+    // try to build a tx xdr, if you cannot then assume the user wants to sign an arbitrary blob
+    let encodedBlob = ""
+    try {
+      const transaction = SDK.TransactionBuilder.fromXDR(
```

Comment:
Yeah, I like your idea of splitting this up into a new API method called `signBlob`. I think signing a blob is a different enough, and perhaps more of a super-user feature, that we can just have its own dedicated method for it rather than splitting it here on the try/catch

Reply:
Added in f3865d3d3d44002448a251d1a4055d4852bc624f

Reply:
100% agreed: signing arbitrary data is **very** dangerous

Reply:
@Shaptic can you expand your thoughts on this? I'm trying to think through ways to make this safer. I think we can safely decode the b64 blob and display it in the signing window/make the window clearly show that this is not XDR, but can you think of anything else we can do to make this safer for end users?

Reply:
I think a warning around "Do not sign random blobs unless you know what you're signing/doing, trust their source, etc." would be really helpful! As in "we have no idea what this blob represents so be aware of that"

Reply:
@piyalbasu I added these kind of off the cuff, any thoughts or additions for this warning?
![Screenshot 2023-07-31 at 2 40 58 PM](https://github.com/stellar/freighter/assets/6886006/ee0d02f9-7bc3-4f83-adf3-91481454f316)

![Screenshot 2023-07-31 at 2 41 10 PM](https://github.com/stellar/freighter/assets/6886006/30a4ff52-8f6a-4a06-9285-99f19bb58834)

Reply:
It also shows the blob you are signing underneath

Reply:
Nice - I think this is a great start. We can run this by Bruno to see if he wants to adjust this copy at all